app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -195,15 +211,43 @@ public void onClick(DialogInterface dialog, int which) {
                 }
 
                 // Handle post creation logic
-                try {
-                    // Create the post in the API
-                    int newPostID = API.addPost(itemName, itemLink, currentUser.getId());
-
-                    // Update the post's folder in the API
-                    API.addPostToFolder(newPostID, selectedFolderId);
-                } catch (APIException e) {
-                    throw new RuntimeException(e);
-                }
+                Call<ResponseBody> call = apiService.createPost("Token " + currentUser.getToken(), new PostRequest(itemName, "", itemLink, selectedFolderId));
```

Comment:
It is populated from a spinner lines 199 and 200, I believe. But since we don't allow a post to exist outside a folder, post creation would be better located inside FolderView most likely.

Reply:
I see the spinner above, but I'm not sure where selectedFolderId would've been set. If we're on the dashboard, presumably the user hasn't selected anything yet, no?

I agree, though. I think creating a Folder should exist here, and creating a Post should exist inside a given folder (where the Folder ID will be implicit)